docs/how_to/annotate_ecephys_data.rst

```diff
@@ -0,0 +1,140 @@
+
+Most of the key metadata related to extracellular electrophysiology is stored in the electrodes table of the NWB file.
+The electrode table contains a row per electrode where electrode specific metadata can be stored.
+Two of the most important properties are the ``ElectrodeGroup`` assignments and the anatomical localization of the electrodes, both of which are required by the NWB format.
+
+When annotating the electrodes table, be sure to follow the `best practices for the electrode table in NWB files <https://nwbinspector.readthedocs.io/en/dev/best_practices/ecephys.html#location>`_.
+
+
+How to Set ElectrodeGroup Metadata
+----------------------------------
+
+When working with extracellular electrophysiology data, it can be helpful to segment electrodes into different groups for analysis (e.g., spike sorting).
+Electrodes are typically grouped by physical proximity (e.g., tetrodes, probe shanks) or by probe.
+
+By default, NeuroConv creates a single ElectrodeGroup that includes all channels.
+This guide demonstrates how to define multiple distinct electrode groups and annotate them so that
+NeuroConv correctly assigns them during data conversion.
+
+
+.. code-block:: python
+
+    from neuroconv.tools.testing.mock_interfaces import MockRecordingInterface
+    from neuroconv.tools import configure_and_write_nwbfile
+
+    interface = MockRecordingInterface(num_channels=5, durations=[0.100])
+    metadata = interface.get_metadata()
+
+    # Define multiple electrode groups with different metadata
+    electrode_group_A = {"name": "ElectrodeGroupA", "description": "A description", "location": "Location A"}
+    electrode_group_B = {"name": "ElectrodeGroupB", "description": "B description", "location": "Location B"}
+    electrode_group_C = {"name": "ElectrodeGroupC", "description": "C description", "location": "Location C"}
+
+    # Replace the default ElectrodeGroup with a list of electrode groups
+    electrode_groups = [electrode_group_A, electrode_group_B, electrode_group_C]
+    metadata["Ecephys"]["ElectrodeGroup"] = electrode_groups
+
+    # Map channel ids to group names, note that you can get the channel ids of the interface with:
+    # interface.channel_ids
+    recording = interface.recording_extractor
+    channel_id_to_group_names = {
+        "0": "ElectrodeGroupA",
+        "1": "ElectrodeGroupA",
+        "2": "ElectrodeGroupC",
+        "3": "ElectrodeGroupB",
+        "4": "ElectrodeGroupA",
+    }
+    recording.set_property(
+        key="group",
+        values=list(channel_id_to_group_names.values()),
+        ids=list(channel_id_to_group_names.keys()),
+    )
+
+    # Create the NWBFile with the updated metadata
+    nwbfile = interface.create_nwbfile(metadata=metadata)
+
+    # You can check that your electrodes are correctly representing here
+    nwbfile.electrode_groups
+
+    # And that they are mapped correctly to the channels/electrodes
+    nwbfile.electrodes.to_dataframe()
```

Comment:
I think it would be also useful to show how this table looks like, easier to imagine where the group ends up in the table. but only if you agree as well, this is very minor